client/blocks/importer/wordpress/upgrade-plan/with-migration-sticker.tsx

```diff
@@ -0,0 +1,32 @@
+import config from '@automattic/calypso-config';
+import { useLayoutEffect } from 'react';
+import { useMigrationStickerMutation } from 'calypso/data/site-migration/use-migration-sticker';
+import UpgradePlanLoader from './upgrade-plan-loader';
+import type { UpgradePlanDetailsProps } from './types';
+import type { FC } from 'react';
+
+const withMigrationSticker =
```

Comment:
I started doing it as a wrapper component, but I decided to go with a HOC in this case, which fits better because `UpgradePlanDetails` totally depends on it and it doesn't make sense using the `UpgradePlanDetails` without the wrapper (now as a HOC).

HOCs are not commonly used anymore, but in this case that we want to avoid the other hooks being called until the migration sticker is added, it will work better.

Commit changing it to a HOC: https://github.com/Automattic/wp-calypso/pull/92147/commits/7ce732245e3816b16bbdd5bb7923beb48174e475